pkg/logql/log/parser.go

```diff
@@ -64,6 +65,10 @@ func (j *JSONParser) Process(_ int64, line []byte, lbs *LabelsBuilder) ([]byte,
 	j.lbs = lbs
 
 	if err := jsonparser.ObjectEach(line, j.parseObject); err != nil {
+		if errors.Is(err, errFoundAllLabels) {
+			return line, true
+		}
```

Comment:
Ah, this is the old trick to use "exceptions". I think this needs a comment that it is not an actually error :slightly_smiling_face: